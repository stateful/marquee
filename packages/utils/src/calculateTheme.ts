import { createTheme, ThemeOptions } from '@mui/material/styles'
import React from 'react'

export default () => {
  const fontSize = parseInt(
    window.getComputedStyle(document.documentElement)
      .getPropertyValue('--vscode-font-size')
      .replace('px', '')
  )

  const vscodeStyles = document.documentElement.style.cssText.split(';')
  const vsCodeStyleMap: Record<string, string> = {}

  vscodeStyles.forEach((style) => {
    if (style.indexOf('--vscode') !== -1) {
      const pair = style.replace(/\s/g, '').split(':')
      vsCodeStyleMap[pair[0]] = pair[1]
    }
  })

  const newTheme: ThemeOptions = {
    shape: {
      borderRadius: 0,
    },
    components: {
      MuiSvgIcon: {
        defaultProps: {
          htmlColor: vsCodeStyleMap['--vscode-icon-foreground']
        }
      },
      MuiChip: {
        styleOverrides: {
          root: {
            borderRadius: '0px',
          }
        }
      },
      MuiButton: {
        styleOverrides: {
          root: {
            color: 'var(--vscode-button-foreground)',
            backgroundColor: 'var(--vscode-button-background)',
            borderColor: 'var(--vscode-editorGhostText-foreground)',
            '&:hover': {
              backgroundColor: 'var(--vscode-button-hoverBackground)',
              borderColor: 'var(--vscode-tab-inactiveForeground)'
            }
          },
          outlined: {
            backgroundColor: 'transparent',
            color: 'var(--vscode-button-background)',
          }
        }
      },
      MuiListItem: {
        styleOverrides: {
          root: {
            '&:hover': {
              backgroundColor: 'var(--vscode-button-secondaryHoverBackground)',
              color: 'var(--vscode-button-secondaryForeground)'
            }
          }
        }
      },
      MuiButtonGroup: {
        styleOverrides: {
          grouped: {
            '&:not(:last-of-type)': {
              borderRight: '1px solid rgba(255, 255, 255, 0.23)'
            }
          }
        }
      },
      MuiDialog: {
        styleOverrides: {
          paper: {
            backgroundImage: 'none'
          }
        }
      },
      MuiPopover: {
        styleOverrides: {
          paper: {
            backgroundImage: 'none'
          }
        }
      },
      MuiTabs: {
        styleOverrides: {
          indicator: {
            left: 0
          }
        }
      }
    },
    palette: {
      mode: document.body.classList[0] === 'vscode-light' ? 'light' : 'dark',
      divider: vsCodeStyleMap['--vscode-foreground'],
      background: {
        default: vsCodeStyleMap['--vscode-sideBar-background'],
        paper: vsCodeStyleMap['--vscode-sideBar-background'],
      },
      text: {
        primary: vsCodeStyleMap['--vscode-foreground'],
        secondary: vsCodeStyleMap['--vscode-foreground'],
      },
      primary: {
        main: vsCodeStyleMap['--vscode-button-background'] || '#000000',
        dark: vsCodeStyleMap['--vscode-button-hoverBackground'],
        contrastText: vsCodeStyleMap['--vscode-button-foreground'],
      },
      secondary: {
        main: vsCodeStyleMap['--vscode-editorMarkerNavigationError-background'],
        contrastText: vsCodeStyleMap['--vscode-button-foreground'],
      },
      action: {
        active: vsCodeStyleMap['--vscode-foreground'],
        selected: vsCodeStyleMap['--vscode-editor-selectionBackground'],
        hover: vsCodeStyleMap['--vscode-editor-hoverHighlightBackground'],
        disabled: vsCodeStyleMap['--vscode-editor-inactiveSelectionBackground'],
        disabledBackground:
          vsCodeStyleMap['--vscode-editor-inactiveSelectionBackground'],
      },
    },
    typography: {
      fontFamily: vsCodeStyleMap['--vscode-font-family'],
      fontWeightLight: vsCodeStyleMap['--vscode-font-weight'] as React.CSSProperties['fontWeight'],
      fontWeightRegular: vsCodeStyleMap['--vscode-font-weight'] as React.CSSProperties['fontWeight'],
      fontWeightMedium: vsCodeStyleMap['--vscode-font-weight'] as React.CSSProperties['fontWeight'],
      fontWeightBold: vsCodeStyleMap['--vscode-font-weight'] as React.CSSProperties['fontWeight'],
      fontSize: fontSize,
      button: {
        textTransform: 'none',
      },
    }
  }

  return createTheme(newTheme)
}
