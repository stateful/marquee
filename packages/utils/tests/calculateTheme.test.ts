import calculateTheme from '../src/calculateTheme'

test('calculateTheme', () => {
  expect(calculateTheme()).toMatchSnapshot()
})
