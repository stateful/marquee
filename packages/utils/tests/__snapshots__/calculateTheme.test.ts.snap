// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateTheme 1`] = `
Object {
  "breakpoints": Object {
    "between": [Function],
    "down": [Function],
    "keys": Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ],
    "not": [Function],
    "only": [Function],
    "unit": "px",
    "up": [Function],
    "values": Object {
      "lg": 1200,
      "md": 900,
      "sm": 600,
      "xl": 1536,
      "xs": 0,
    },
  },
  "components": Object {
    "MuiButton": Object {
      "styleOverrides": Object {
        "root": Object {
          "&:hover": Object {
            "backgroundColor": "var(--vscode-tree-tableOddRowsBackground)",
            "borderColor": "var(--vscode-tab-inactiveForeground)",
            "color": "rgb(204, 204, 204)",
          },
          "borderColor": "var(--vscode-editorGhostText-foreground)",
          "color": "rgb(204, 204, 204)",
        },
      },
    },
    "MuiButtonGroup": Object {
      "styleOverrides": Object {
        "grouped": Object {
          "&:not(:last-of-type)": Object {
            "borderRight": "1px solid rgba(255, 255, 255, 0.23)",
          },
        },
      },
    },
    "MuiChip": Object {
      "styleOverrides": Object {
        "root": Object {
          "borderRadius": "0px",
        },
      },
    },
    "MuiDialog": Object {
      "styleOverrides": Object {
        "paper": Object {
          "backgroundImage": "none",
        },
      },
    },
    "MuiPopover": Object {
      "styleOverrides": Object {
        "paper": Object {
          "backgroundImage": "none",
        },
      },
    },
    "MuiSvgIcon": Object {
      "defaultProps": Object {
        "htmlColor": "#c5c5c5",
      },
    },
    "MuiTabs": Object {
      "styleOverrides": Object {
        "indicator": Object {
          "left": 0,
        },
      },
    },
  },
  "direction": "ltr",
  "mixins": Object {
    "toolbar": Object {
      "@media (min-width:0px)": Object {
        "@media (orientation: landscape)": Object {
          "minHeight": 48,
        },
      },
      "@media (min-width:600px)": Object {
        "minHeight": 64,
      },
      "minHeight": 56,
    },
  },
  "palette": Object {
    "action": Object {
      "activatedOpacity": 0.24,
      "active": "#cccccc",
      "activeChannel": "255 255 255",
      "disabled": "#3a3d41",
      "disabledBackground": "#3a3d41",
      "disabledOpacity": 0.38,
      "focus": "rgba(255, 255, 255, 0.12)",
      "focusOpacity": 0.12,
      "hover": "rgba(38,79,120,0.25)",
      "hoverOpacity": 0.08,
      "selected": "#264f78",
      "selectedOpacity": 0.16,
    },
    "augmentColor": [Function],
    "background": Object {
      "default": "#252526",
      "paper": "#252526",
    },
    "common": Object {
      "black": "#000",
      "white": "#fff",
    },
    "contrastThreshold": 3,
    "divider": "#cccccc",
    "error": Object {
      "contrastText": "#fff",
      "dark": "#d32f2f",
      "light": "#e57373",
      "main": "#f44336",
    },
    "getContrastText": [Function],
    "grey": Object {
      "100": "#f5f5f5",
      "200": "#eeeeee",
      "300": "#e0e0e0",
      "400": "#bdbdbd",
      "50": "#fafafa",
      "500": "#9e9e9e",
      "600": "#757575",
      "700": "#616161",
      "800": "#424242",
      "900": "#212121",
      "A100": "#f5f5f5",
      "A200": "#eeeeee",
      "A400": "#bdbdbd",
      "A700": "#616161",
      "dark": "#616161",
      "darkChannel": "97 97 97",
    },
    "info": Object {
      "contrastText": "rgba(0, 0, 0, 0.87)",
      "dark": "#0288d1",
      "light": "#4fc3f7",
      "main": "#29b6f6",
    },
    "mode": "dark",
    "primary": Object {
      "contrastText": "#ffffff",
      "dark": "#1177bb",
      "light": "rgb(62, 130, 175)",
      "main": "#0e639c",
    },
    "secondary": Object {
      "contrastText": "#ffffff",
      "dark": "rgb(168, 53, 53)",
      "light": "rgb(243, 111, 111)",
      "main": "#f14c4c",
    },
    "success": Object {
      "contrastText": "rgba(0, 0, 0, 0.87)",
      "dark": "#388e3c",
      "light": "#81c784",
      "main": "#66bb6a",
    },
    "text": Object {
      "disabled": "rgba(255, 255, 255, 0.5)",
      "icon": "rgba(255, 255, 255, 0.5)",
      "primary": "#cccccc",
      "primaryChannel": "255 255 255",
      "secondary": "#cccccc",
      "secondaryChannel": "255 255 255",
    },
    "tonalOffset": 0.2,
    "warning": Object {
      "contrastText": "rgba(0, 0, 0, 0.87)",
      "dark": "#f57c00",
      "light": "#ffb74d",
      "main": "#ffa726",
    },
  },
  "shadows": Array [
    "none",
    "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
    "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
    "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
    "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
    "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
    "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
    "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
    "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
    "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
    "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
    "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
    "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
    "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
    "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
    "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
    "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
    "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
    "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
  ],
  "shape": Object {
    "borderRadius": 0,
  },
  "spacing": [Function],
  "transitions": Object {
    "create": [Function],
    "duration": Object {
      "complex": 375,
      "enteringScreen": 225,
      "leavingScreen": 195,
      "short": 250,
      "shorter": 200,
      "shortest": 150,
      "standard": 300,
    },
    "easing": Object {
      "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
      "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
      "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
      "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
    },
    "getAutoHeightDuration": [Function],
  },
  "typography": Object {
    "body1": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "0.9285714285714286rem",
      "fontWeight": "normal",
      "lineHeight": 1.5,
    },
    "body2": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "0.8125rem",
      "fontWeight": "normal",
      "lineHeight": 1.43,
    },
    "button": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "0.8125rem",
      "fontWeight": "normal",
      "lineHeight": 1.75,
      "textTransform": "none",
    },
    "caption": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "0.6964285714285714rem",
      "fontWeight": "normal",
      "lineHeight": 1.66,
    },
    "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
    "fontSize": 13,
    "fontWeightBold": "normal",
    "fontWeightLight": "normal",
    "fontWeightMedium": "normal",
    "fontWeightRegular": "normal",
    "h1": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "5.571428571428571rem",
      "fontWeight": "normal",
      "lineHeight": 1.167,
    },
    "h2": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "3.482142857142857rem",
      "fontWeight": "normal",
      "lineHeight": 1.2,
    },
    "h3": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "2.7857142857142856rem",
      "fontWeight": "normal",
      "lineHeight": 1.167,
    },
    "h4": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "1.9732142857142858rem",
      "fontWeight": "normal",
      "lineHeight": 1.235,
    },
    "h5": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "1.3928571428571428rem",
      "fontWeight": "normal",
      "lineHeight": 1.334,
    },
    "h6": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "1.1607142857142858rem",
      "fontWeight": "normal",
      "lineHeight": 1.6,
    },
    "htmlFontSize": 16,
    "overline": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "0.6964285714285714rem",
      "fontWeight": "normal",
      "lineHeight": 2.66,
      "textTransform": "uppercase",
    },
    "pxToRem": [Function],
    "subtitle1": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "0.9285714285714286rem",
      "fontWeight": "normal",
      "lineHeight": 1.75,
    },
    "subtitle2": Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,sans-serif",
      "fontSize": "0.8125rem",
      "fontWeight": "normal",
      "lineHeight": 1.57,
    },
  },
  "zIndex": Object {
    "appBar": 1100,
    "drawer": 1200,
    "fab": 1050,
    "mobileStepper": 1000,
    "modal": 1300,
    "snackbar": 1400,
    "speedDial": 1050,
    "tooltip": 1500,
  },
}
`;
